Automattic/jetpack
# Sidebar menu: Register "plugin install" route on sites with the default interface

## Proposed changes:

In https://github.com/Automattic/jetpack/pull/37521 we added a new "Add New Plugin" submenu which is only registered on sites with the classic admin interface so we could redirect from `/wp-admin/plugin-install.php` to the route of that submenu (`/wp-admin/plugins.php?page=wpcom-install-plugin`).

However, for that redirect to work, we need the new route to be working on all Simple sites, including sites with the default admin interface (see pMz3w-k1d-p2#comment-119194).

Since we don't want the "Add New Plugin" submenu to be visible on sites with the default admin interface, this PR implements a workaround in which it registers the menu (on all sites) and then hides it (on sites with the default interface).

To simplify the workaround, the existing logic that registers the Plugins parent menu from Nav Unification to the jetpack-mu-wpcom package.

### Other information:


## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
- Apply these changes to your WP.com simple site
- With the classic admin interface, make sure you get the same behavior as before (3 submenus: Plugins > Installed Plugins, Plugins > Add New Plugin, Plugins > Marketplace)
- With the default admin interface:
  - Make sure Plugins still links to the Marketplace
  - Make sure there are no submenus
  - Make sure you can visit `/wp-admin/plugins.php?page=wpcom-install-plugin`
